text/0000-import-trait-methods.md

```diff
@@ -0,0 +1,253 @@
+- Feature Name: `import-trait-methods`
+- Start Date: 2024-03-19
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Allow importing methods from traits and then using them like regular functions.
```

Comment:
If this PR only plans on adding support for importing associated *functions*, it should state so explicitly and add a short explainer why that's the case. Note that implementing this feature for associated *constants* would probably be easier to do than for associated *functions*, so there's not much reason to exclude them, imho.

(Importing associated *types* on the other hand should probably be forbidden. Unless we reify the implicit `Self` type parameter, such an associated type `Ty` would be equivalent to `<_ as Trait>::Ty` which is impossible to solve since associated types are not injective in general.)

Reply:
I've only ever seen this feature mentioned in with associated funcs. Writing in associated consts shouldn't be too hard, but I only want to do it if it is wanted.

Reply:
@obsgolem If you think associated constants would be straightforward to add to the RFC, please do go ahead and add it. If you think it'd be a challenge to add to the RFC, please mention it in future work instead.